src/AppInstallerRepositoryCore/Rest/Schema/Json/InformationResponseDeserializer.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "Rest/Schema/IRestClient.h"
+#include <cpprest/json.h>
+#include "JsonHelper.h"
+#include "InformationResponseDeserializer.h"
+#include "CommonRestConstants.h"
+
+namespace AppInstaller::Repository::Rest::Schema::Json
+{
+    namespace
+    {
+        // Information response constants
+        constexpr std::string_view SourceIdentifier = "SourceIdentifier"sv;
+        constexpr std::string_view ServerSupportedVersions = "ServerSupportedVersions"sv;
+        constexpr std::string_view APIVersion = "APIVersion"sv;
+        constexpr std::string_view ObjectVersions = "ObjectVersions"sv;
+    }
+
+    std::optional<IRestClient::Information> InformationResponseDeserializer::Deserialize(const web::json::value& dataObject) const
+    {
+        // Get information result from json output.
+        std::optional<std::reference_wrapper<const web::json::value>> data = JsonHelper::GetJsonValueFromNode(dataObject, JsonHelper::GetJsonKeyNameString(Data));
+        if (!data.has_value())
+        {
+            AICLI_LOG(Repo, Verbose, << "Missing data");
+            return {};

```

Comment:
General error handling strategy for these deserialization methods:
1. Log as Error
2. Returning an empty optional is fine *as long as* the caller treats that as an error and throws an appropriate error understandable to our stack.
3. Wrap in a function level try/catch that also returns an empty optional.
4. I would suggest combining 2 and 3 by making the public method non-optional and throw and a private one that does the work.

Reply:
Basically, don't ignore error conditions by just skipping them and also don't propagate the REST specific exception types out of the source.

If a value is truly optional, then you wouldn't return early.

Reply:
Have added the errors and made public methods throw when it gets an optional value.

---
In reply to: [596389291](https://github.com/microsoft/winget-cli/pull/800#discussion_r596389291) [](ancestors = 596389291)